algorithms/sorting/quicksort.py

```diff
@@ -0,0 +1,47 @@
+"""Module with the implementation of the QuickSort algorithm."""
+
+
+from typing import List
+from random import randint
+
+from .sort import Sort
+
+
+class QuickSort(Sort):
+    """Class that represents a QuickSort implementation."""
+
+    def _sort(self, items: List) -> List:
+        size = len(items)
+        sorted_items = items.copy()
+
+        self.__sort(sorted_items, 0, size - 1)
+        return sorted_items
+
+    def __sort(self, items: List, p: int, r: int) -> None:
```

Comment:
Is this a normal naming convention in `Python`? It looks like it is, but I don't know what the standard implication of naming a method starting with double underscores `__` means. I think we should consider the alternative of using the `@typing.overload` decorator to name the method the same way as the original, but have the opportunity of changing the parameters so we can perform the `QuickSort` the way we want.

Reply:
It's to make the method private, so only the method defined in the `Sort` class is used, that's the idea of all other sorting algorithms. But we can also overload the `sort method` to use both.

Reply:
I'm ok with both options. Go with the one you consider the best. Thanks.

Reply:
I think that if we want to use this method to sort a subarray, it's better to add this functionality to all the other sorting algorithms, so that we have something uniform in all the algorithms.